python_modules/libraries/dagster-spark/dagster_spark/solids.py

```diff
@@ -8,6 +8,64 @@
 from .utils import run_spark_subprocess, parse_spark_config
 
 
+def create_spark_shell_cmd(solid_config, main_class):
+    # Extract parameters from config
+    (master_url, deploy_mode, application_jar, spark_conf, application_arguments, spark_home) = [
+        solid_config.get(k)
+        for k in (
+            'master_url',
+            'deploy_mode',
+            'application_jar',
+            'spark_conf',
+            'application_arguments',
+            'spark_home',
+        )
+    ]
+
+    # Let the user use env vars in the jar path
+    application_jar = os.path.expandvars(application_jar)
+
+    if not os.path.exists(application_jar):
+        raise SparkSolidError(
+            (
+                'Application jar {} does not exist. A valid jar must be '
+                'built before running this solid.'.format(application_jar)
+            )
+        )
+
+    spark_home = spark_home if spark_home else os.environ.get('SPARK_HOME')
+
+    if spark_home is None:
+        raise SparkSolidError(
+            (
+                'No spark home set. You must either pass spark_home in config or '
+                'set $SPARK_HOME in your environment (got None).'
+            )
+        )
+
+    deploy_mode = ['--deploy-mode', '{}'.format(deploy_mode)] if deploy_mode else []
+
+    spark_shell_cmd = (
+        ['{}/bin/spark-submit'.format(spark_home), '--class', main_class, '--master', master_url]
+        + deploy_mode
+        + parse_spark_config(spark_conf)
+        + [application_jar]
+        + ([application_arguments] if application_arguments else [])
+    )
+    return spark_shell_cmd
+
+
+def create_step_metadata_fn(solid_name, main_class):
```

Comment:
just a style difference to consider, but thoughts on something like:


```
def step_metadata_fn(solid_name, main_class, environment_config):
    return {
        'spark_submit_command': ' '.join(
            create_spark_shell_cmd(environment_config.solids[solid_name].config, main_class)
        )

step_metadata_fn=functools.partial(step_metadata_fn, solid_name=solid_name, main_class=main_class)
```

Reply:
into it